dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java

```diff
@@ -357,4 +365,66 @@ public void testingTimeMachine () throws Exception{
         }
     }
+
+    /**
+     * Here were testing that a logged in user with FE and BE roles will But login in the Front end will get the Page Per SE and not the Edit Mode
+     * More context here: https://github.com/dotcms/core/issues/22124
+     * @throws Exception
+     */
+    @Test
+    public void Test_Frontend_Login_Overrides_BackendLogin_When_Page_Requested() throws Exception {
+
+        final User loginUser = mock(User.class);
+        when(loginUser.hasConsoleAccess()).thenReturn(true);
+        when(loginUser.isAnonymousUser()).thenReturn(false);
+        when(loginUser.isBackendUser()).thenReturn(true);
+        when(loginUser.isFrontendUser()).thenReturn(true);
+        when(loginUser.isActive()).thenReturn(true);
+
+        testServerPageFor(loginUser, LoginMode.FE);
+        testServerPageFor(loginUser, LoginMode.BE);
+        testServerPageFor(loginUser, LoginMode.UNKNOWN);
+
+    }
+
+    /**
+     * This is the actual test body
+     * @param user
+     * @param mode
+     * @throws IOException
+     * @throws DotSecurityException
+     * @throws DotDataException
+     * @throws ServletException
+     */
+    private void testServerPageFor(final User user, final LoginMode mode)
+            throws IOException, DotSecurityException, DotDataException, ServletException {
+        final String pageContent = "<html>lol</html>";
+
+        VelocityRequestWrapper velocityRequest = mock(VelocityRequestWrapper.class);
+        when(velocityRequest.getRequestURI()).thenReturn("/lol");
+        when(velocityRequest.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(user);
+        final HttpSession session = mock(HttpSession.class);
+        when(session.getAttribute(LOGIN_MODE_PARAMETER)).thenReturn(mode);
+        when(velocityRequest.getSession(Mockito.anyBoolean())).thenReturn(session);
+        when(velocityRequest.getSession()).thenReturn(session);
+
+        final ServletOutputStream outputStream = mock(ServletOutputStream.class);
+        when(response.getOutputStream()).thenReturn(outputStream);
+
+        final HTMLPageAssetRenderedAPI pageAssetRenderedAPI = mock(HTMLPageAssetRenderedAPIImpl.class);
+        when(pageAssetRenderedAPI.getPageHtml(Mockito.any(PageContext.class),Mockito.any(HttpServletRequest.class), Mockito.any(HttpServletResponse.class))).thenReturn(
+                pageContent);
+
+        final VelocityServlet velocityServlet = new VelocityServlet(WebAPILocator.getUserWebAPI(), pageAssetRenderedAPI);
+        velocityServlet.service(velocityRequest, response);
+        if(LoginMode.FE == mode){
+            //Here we verify the page was served normally
+            verify(outputStream, times(1)).write(pageContent.getBytes());
+        } else {
+            //Here the page never got served and a redirect occurred taking the user to EditMode
+            verify(outputStream, never()).write(pageContent.getBytes());
```

Comment:
but in this case is not called but with the page's EDIT vesion?

Reply:
when EDIT Mode is "called" or whatever it's redirect therefore the outputstream write never gets called